Classes/System/Records/Pages/PagesRepository.php

```diff
@@ -167,28 +167,30 @@ public function findAllSubPageIdsByRootPage(
             return $this->transientVariableCache->get($cacheIdentifier);
         }
 
-        $pageIdsList = $this->getTreeList($rootPageId, 9999, 0, 'deleted = 0');
+        $permClause = empty($initialPagesAdditionalWhereClause) ? 'deleted = 0' : ('deleted = 0 AND ' . $initialPagesAdditionalWhereClause);
+        $pageIdsList = $this->getTreeList($rootPageId, 9999, 0, $permClause);
         $pageIds = GeneralUtility::intExplode(',', $pageIdsList);
 
         if (!empty($initialPagesAdditionalWhereClause)) {
-            $pageIds = $this->filterPageIdsByInitialPagesAdditionalWhereClause($pageIds, $initialPagesAdditionalWhereClause);
+            $pageIds = $this->filterPageIdsByAdditionalWhereClause($pageIds, $initialPagesAdditionalWhereClause);
```

Comment:
On this place the `$initialPagesAdditionalWhereClause` MUST NOT be used,
the value of `plugin.tx_solr.index.queue.[indexConfig].additionalWhereClause`.

Like in the docs described:
[plugin.tx_solr.index.queue.[indexConfig].additionalWhereClause
](https://docs.typo3.org/p/apache-solr-for-typo3/solr/main/en-us/Configuration/Reference/TxSolrIndex.html?highlight=initialpagesadditionalwhereclause#queue-indexconfig-additionalwhereclause)+
[plugin.tx_solr.index.queue.[indexConfig].initialPagesAdditionalWhereClause](https://docs.typo3.org/p/apache-solr-for-typo3/solr/main/en-us/Configuration/Reference/TxSolrIndex.html?highlight=initialpagesadditionalwhereclause#queue-indexconfig-additionalwhereclause)